### PR TITLE
[flang][OpenMP] Fix bug in emitting `dealloc` logic

### PR DESCRIPTION
**Pull request by ergawy**
Fixes a bug in emiting deacllocation logic when delayed privatization is disabled. I introduced the bug when implementing delayed privatization for allocatables: when delayed privatization is disabled the deacllocation ops are emitted for only one allocatable variables.

### CODE REVIEW

**Review by NimishMishra — COMMENTED** [review]
Thanks for the fix. The change looks good.

**Comment by kiranchandramohan** [comment]
> Thanks for the fix. The change looks good.

Please approve if it is fine.

File: flang/test/Lower/OpenMP/allocatable-multiple-vars.f90 [hunk 1]
```diff
@@ -0,0 +1,28 @@
+! Test delayed privatization for multiple allocatable variables.
```
Comment (kiranchandramohan) [inline flang/test/Lower/OpenMP/allocatable-multiple-vars.f90:1]:
```suggestion
! Test early privatization for multiple allocatable variables.
```

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Pull request merged**